iceoryx_utils/test/moduletests/test_cxx_forward_list.cpp

```diff
@@ -0,0 +1,1931 @@
+    EXPECT_TRUE((++iter) == sut.cend());
+}
+
+TEST_F(forward_list_test, ConstIteratorIncrementOperatorBeyondEnd)
+{
+    constexpr int DEFAULT_VALUE{13};
+
+    // fill
+    sut.push_front(DEFAULT_VALUE);
+
+    auto iter = sut.cbegin();
+    EXPECT_THAT(iter->m_value, Eq(DEFAULT_VALUE));
+    EXPECT_TRUE((++iter) == sut.cend());
+    EXPECT_TRUE((++iter) == sut.cend());
+}
+
+
+TEST_F(forward_list_test, IteratorComparisonOfDifferentLists)
+{
+    forward_list<CTorTest, TESTLISTCAPACITY> cut1, cut2;
+    cut1.emplace_front(15842);
+    cut1.emplace_front(1584122);
+    cut1.emplace_front(158432);
+    cut1.emplace_front(158432);
+
+    cut2.emplace_front(1313);
+    cut2.emplace_front(13131);
+
+
+    auto iter_sut1 = cut1.begin();
+    auto iter_sut2 = cut2.begin();
+    EXPECT_DEATH(dummyFunc(iter_sut1 == iter_sut2), "");
+
+    iter_sut1 = cut1.before_begin();
+    iter_sut2 = cut2.before_begin();
+    EXPECT_DEATH(dummyFunc(iter_sut1 == iter_sut2), "");
+
+    iter_sut1 = cut1.end();
+    iter_sut2 = cut2.end();
+    EXPECT_DEATH(dummyFunc(iter_sut1 == iter_sut2), "");
+
+    iter_sut1 = cut1.begin();
+    iter_sut2 = cut2.begin();
+    EXPECT_DEATH(dummyFunc(iter_sut1 != iter_sut2), "");
+
+    iter_sut1 = cut1.before_begin();
+    iter_sut2 = cut2.before_begin();
+    EXPECT_DEATH(dummyFunc(iter_sut1 != iter_sut2), "");
+
+    iter_sut1 = cut1.end();
+    iter_sut2 = cut2.end();
+    EXPECT_DEATH(dummyFunc(iter_sut1 != iter_sut2), "");
+}
+
+
+TEST_F(forward_list_test, ComparingConstIteratorAndIterator)
+{
+    forward_list<CTorTest, TESTLISTCAPACITY> cut1, cut2;
+    cut1.emplace_front(15842);
+    cut1.emplace_front(1584122);
+    cut1.emplace_front(158432);
+    cut1.emplace_front(158432);
+
+    cut2.emplace_front(1313);
+    cut2.emplace_front(13131);
+
+
+    forward_list<CTorTest, TESTLISTCAPACITY>::const_iterator iter_sut1 = cut1.cbefore_begin();
+    forward_list<CTorTest, TESTLISTCAPACITY>::const_iterator iter_sut2 = cut1.cbefore_begin();
+    forward_list<CTorTest, TESTLISTCAPACITY>::iterator iter_sut3 = cut1.begin();
+    forward_list<CTorTest, TESTLISTCAPACITY>::iterator iter_sut4 = cut1.end();
+
+    ASSERT_THAT(iter_sut1 == iter_sut3, Eq(false));
+    ASSERT_THAT(iter_sut3 == iter_sut1, Eq(false));
+
+    ASSERT_THAT(iter_sut1 == iter_sut2, Eq(true));
+    ASSERT_THAT(iter_sut4 == iter_sut3, Eq(false));
+}
+
+
+TEST_F(forward_list_test, IteratorTraitsGetValueType)
+{
+    forward_list<int, 10> cut;
+
+    cut.emplace_front(5);
+    auto iter{cut.begin()};
+
+    // using a function call here is closer to the actual use case (-> intentionally did not inline all code here)
+    auto ret = iteratorTraitReturnDoubleValue(iter);
+
+    EXPECT_THAT(ret, Eq(10));
+}
+
+TEST_F(forward_list_test, IteratorTraitsCheckIteratorCategoryOnConstIterator)
+{
+    int cnt = 0;
+    auto iter = sut.cbefore_begin();
+    if (typeid(std::iterator_traits<decltype(iter)>::iterator_category) == typeid(std::random_access_iterator_tag))
```

Comment:
Could have used ASSERT_EQ without if or cnt. The point of the test is to check that the iterator_category is correct I suppose, worth testing but I think it could be simplified.